docs/src/examples/basics.md

```diff
@@ -0,0 +1,141 @@
+# Basics
+
+## Basic operations for quaternions
+Quaternions can be defined with [`Quaternion`](@ref) constructor or [`quat`](@ref) function.
+Note that the order of the arguments is ``w+xi+yj+zk``, not ``xi+yj+zk+w``.
+
+```@repl intro
+using Quaternions
+q1 = Quaternion(1,2,3,4)
+q2 = quat(5,6,7,8)
+q3 = quat(9)
+```
+
+The multiplication is not commutative.
+```@repl intro
+q1 * q2
+q2 * q1
+```
+
+The multiplicative inverse can be calculated with [`Base.inv`](@ref).
+```@repl intro
+inv(q1)
+inv(q1) * q1
+```
+
+The division is also not commutative.
+
+```@repl intro
+q1 / q2  # Same as `q1*inv(q2)` mathematically.
+q2 \ q1  # Same as `inv(q2)*q1` mathematically.
+```
+
+A conjugate of a quaternion can be calculated with [`Base.conj`](@ref).
+But `Base.imag(::Quaternion)` is not defined because it should return three real values which is not consistent with `imag(::Complex)` and `imag(::Real)`.
+Instead, [`imag_part`](@ref) function can be used to obtain the imaginary part of a quaternion.
+See [issue#61](https://github.com/JuliaGeometry/Quaternions.jl/issues/61) for more discussion.
+
+```@repl intro
+conj(q1)
+imag(q1)  # Not supported.
+imag_part(q1)  # Use this instead.
+```
+
+## `Quaternion` vs `quat`
+The general rule is that [`quat`](@ref) is to [`Quaternion`](@ref) as [`complex`](https://docs.julialang.org/en/v1/base/numbers/#Base.complex-Tuple{Complex}) is to [`Complex`](https://docs.julialang.org/en/v1/base/numbers/#Base.Complex).
+`Complex` and `Quaternion` are both constructors so should return an object of the corresponding type, whereas `quat` and `complex` both can operate on types and arrays.
+
+```@setup Quaternion-quat
+using Quaternions
+```
+
+```@repl Quaternion-quat
+Quaternion(1,2,3,4)
+quat(1,2,3,4)
+Quaternion(Int)  # Similar to `Complex(Int)`.
+quat(Int)  # Similar to `complex(Int)`.
+```
+
+## The type parameter `T` in `Quaternion{T}`
+
+The type parameter `T <: Real` in `Quaternion{T}` represents the type of real and imaginary parts of a quaternion.
+
+### Lipschitz quaternions
```

Comment:
Most of these quaternion types are niche and probably not of interest to nearly all of our users. So perhaps it would be better to not include them in `basics.md` and instead put them in a docs page devoted to specialized quaternion types.

Reply:
I added `type_parameter.md` page.